metavariable-comparison: Fix order of re.match arguments

IagoAbal opened the pull request:
The pattern is the first argument, the string is the second!
See https://docs.python.org/3/library/re.html#re.match

Hoping that this won't break many rules since I expect most users will
be using `metavariable-regex` for this.

Related to #4419 (potential workaround)

test plan:
make test # tests updated

PR checklist:
- [x] Documentation is up-to-date
- [x] Changelog is up-to-date
- [x] Change has no security implications (otherwise, ping security team)

emjin commented:
Do we still need to expose this? If few people are using it, maybe we can deprecate instead?

IagoAbal commented:
> Do we still need to expose this? If few people are using it, maybe we can deprecate instead?

That's a good discussion to have, although in the meantime I would just fix it. We could maybe take out the changelog entry if we don't want to give it publicity.

IagoAbal committed:
metavariable-comparison: Fix order of re.match arguments

The pattern is the first argument, the string is the second!
See https://docs.python.org/3/library/re.html#re.match

Hoping that this won't break many rules since I expect most users will
be using `metavariable-regex` for this.

Related to #4419 (potential workaround)

test plan:
make test # tests updated

IagoAbal commented:
I did delete the changelog entry so we avoid somewhat encouraging users to use this.
